providers/azure.go

```diff
@@ -361,6 +369,49 @@ func (p *AzureProvider) getEmailFromProfileAPI(ctx context.Context, accessToken
 	return getEmailFromJSON(json)
 }
 
+func getUserFromJSON(json *simplejson.Json) (string, error) {
+	var user string
+	var err error
+
+	userEmail, err := json.Get("userPrincipalName").String()
+	user = strings.Split(userEmail, "@")[0]
```

Comment:
Are we sure this is the best place to get the user from? Does this effectively mean that the `user` is just the prefix of `email`? Perhaps there are other claims that should be considered?

Also, I don't think there's anything that prevents `user` from containing an email, maybe we shouldn't mutate values we get from the IDP?

Reply:
so in my Azure instance, the username is appended by "@domain.com" and represented as `userprincipalname`, whereas email is the actual mailbox associated with that user. I've seen the same across a couple of other Azure instances too. My rational behind removing the "@domain.com" is this: since the domain is already represented in `email`, it should be used as an identifier within multiple tenants, but `user` will be limited to only one tenant. not sure if this is in-line with other field extractions within oauth2-proxy